include/amici/vector.h

```diff
@@ -190,15 +208,15 @@ class AmiVectorArray {
       * @return pointer to data array
       */
     realtype *data(int pos) {
-        return vec_array.at(static_cast<decltype(vec_array)::size_type>(pos)).data();
+        return vec_array.at(pos).data();
     }
 
     /** const accessor to data of AmiVector elements
       * @param pos index of AmiVector
       * @return const pointer to data array
       */
     const realtype *data(int pos) const {
-        return vec_array.at(static_cast<decltype(vec_array)::size_type>(pos)).data();
```

Comment:
why is it only complaining in this snippet tho, we use ints to index std::vector literally everywhere.

Reply:
It does complain everywhere. I only stopped putting the casts at some point, because I wasn't too convinced it's worth it... Still not quite sure ;)

Reply:
what about using `size_t` instead of `decltype(vec_array)::size_type`? But I agree, I am not sure how well time and energy is spent changing this everywhere ...

Reply:
The two don't necessarily match. The unsigned index types in the STL are a major annoyance... Leave as is. If at some point we change to multi-dimensional spans or something, this might be gone...